source/quick-reference.txt

```diff
@@ -0,0 +1,570 @@
+       | `API Documentation <{+api-root+}/M_MongoDB_Driver_IMongoCollection_1_UpdateMany.htm>`__
+       | :ref:`Usage Example <csharp-update-many>`
+       | :ref:`Fundamentals <csharp-update-operation>`
+
+     - .. code-block:: csharp
+          :copyable: true
+
+          var filter = Builders<Restaurant>.Filter
+            .Eq(restaurant => restaurant.Cuisine, "Pizza");
+
+          var update = Builders<Restaurant>.Update
+            .Set(restaurant => restaurant.Cuisine, "Pasta and breadsticks");
+
+          var result = _restaurantsCollection.UpdateMany(filter, update);
+
+   * - | **Update Multiple Documents (Async)**
+       |
+       | `API Documentation <{+api-root+}/M_MongoDB_Driver_IMongoCollection_1_UpdateManyAsync.htm>`__ 
+       | :ref:`Usage Example <csharp-update-many>`
+       | :ref:`Fundamentals <csharp-update-operation>`
+
+     - .. code-block:: csharp
+          :copyable: true
+
+          var filter = Builders<Restaurant>.Filter
+            .Eq(restaurant => restaurant.Cuisine, "Pizza");
+
+          var update = Builders<Restaurant>.Update
+            .Set(restaurant => restaurant.Cuisine, "Pasta and breadsticks");
+
+          return await = _restaurantsCollection.UpdateManyAsync(filter, update);
+
+   * - | **Update an Array in a Document**
+       |
+       | `API Documentation <{+api-root+}/M_MongoDB_Driver_IMongoCollection_1_UpdateOne.htm>`__
+       | :ref:`Fundamentals <csharp-update-operation>`
+
+     - .. code-block:: csharp
+          :copyable: true
+
+          var filter = Builders<Restaurant>.Filter
+            .Eq(restaurant => restaurant.Name, "Bagels N Buns");
+
+          var update = Builders<Restaurant>.Update
+            .Push(restaurant => restaurant.Grades, new BsonDocument()
+            {
+              { "date" , DateTime.Now.ToUniversalTime()},
+              { "grade" , "A"}, 
+              { "score", 10}
+            });
+
+          var result = _restaurantsCollection.UpdateOne(filter, update);
+             
+   * - | **Replace a Document**
+       |
+       | `API Documentation <{+api-root+}/M_MongoDB_Driver_IMongoCollection_1_ReplaceOne.htm>`__
+       | :ref:`Usage Example <csharp-replace-one>`
+       | :ref:`Fundamentals <csharp-replace-operation>`  
+
+     - .. code-block:: csharp
+          :copyable: true
+
+          var filter = Builders<Restaurant>.Filter
+            .Eq(restaurant => restaurant.Cuisine, "Pizza");
+
+          // Find ID of first pizza restaurant
+          var oldPizzaRestaurant = _restaurantsCollection.Find(filter).First();
+          var oldId = oldPizzaRestaurant.Id;
+
+          Restaurant newPizzaRestaurant = new()
+          {
+              Id = oldId,
+              Name = "Mongo's Pizza",
+              Cuisine = "Pizza",
+              Address = new BsonDocument
+              {
+                  {"street", "Pizza St"},
+                  {"zipcode", "10003"},
+              },
+              Borough = "Manhattan",
+          };
+
```

Comment:
This change, along with the above Grade class, will also have to be made in other PRs. For ease and reusability, I'd like to make these changes once all the docs have been reassembled, Avengers-like, after review. I'll make a separate ticket to be completed before 1/10.

Please let me know if the extra classes in this PR are satisfactory, since I'll use them as the template across all C# pages.

Reply:
The domain classes look fine. I haven't seen a standard `Address` representation in C#. It is often very application specific. Sometimes you have multiple address lines as `AddressLine1`, `AddressLine2`, ... Other times you have a `List<AddressLine>`. Other times you have `AddressLine` as a multi-line string. Sometimes there are coordinates. Sometimes there are not. Your current definition is no better nor worse than any other I've seen. :)